File: src/com/theah64/mock_api/models/RouteUpdate.java
package com.theah64.mock_api.models;

/**
 * Created by theapache64 on 10/1/18.
 */
public class RouteUpdate {

    private final String id, key, routeId, method, params, delay, description, defaultResponse,createdAt;

    public RouteUpdate(String id, String key, String routeId, String method, String params, String delay, String description, String defaultResponse, String createdAt) {
        this.id = id;
        this.key = key;
        this.routeId = routeId;
        this.method = method;
        this.params = params;
        this.delay = delay;
        this.description = description;
        this.defaultResponse = defaultResponse;
        this.createdAt = createdAt;
    }

    public String getId() {
        return id;
    }

    public String getKey() {
        return key;
    }

    public String getRouteId() {
        return routeId;
    }

    public String getMethod() {
        return method;
    }

    public String getParams() {
        return params;
    }

    public String getDelay() {
        return delay;
    }

    public String getDescription() {
        return description;
    }

    public String getDefaultResponse() {
        return defaultResponse;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    @Override
    public String toString() {
        return "RouteUpdate{" +
                "id='" + id + '\'' +
                ", key='" + key + '\'' +
                ", routeId='" + routeId + '\'' +
                ", method='" + method + '\'' +
                ", params='" + params + '\'' +
                ", delay='" + delay + '\'' +
                ", description='" + description + '\'' +
                ", defaultResponse='" + defaultResponse + '\'' +
                '}';
    }
}

File: src/com/theah64/mock_api/models/RouteUpdate.kt
package com.theah64.mock_api.models

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/**
 * Created by theapache64 on 10/1/18.
 */
class RouteUpdate(
        val id: String?,
        val key: String,
        val routeId: String,
        val method: String,
        val params: String,
        val delay: String,
        val description: String,
        val defaultResponse: String,
        createdAt: Long
) {
    val createdAt: String?

    init {
        this.createdAt = if (createdAt != -1L) READABLE_DATE_FORMAT.format(Date(createdAt * 1000)) else null
    }

    override fun toString(): String {
        return "RouteUpdate{" +
                "id='" + id + '\''.toString() +
                ", key='" + key + '\''.toString() +
                ", routeId='" + routeId + '\''.toString() +
                ", method='" + method + '\''.toString() +
                ", params='" + params + '\''.toString() +
                ", delay='" + delay + '\''.toString() +
                ", description='" + description + '\''.toString() +
                ", defaultResponse='" + defaultResponse + '\''.toString() +
                '}'.toString()
    }

    companion object {

        private val READABLE_DATE_FORMAT = SimpleDateFormat("MMM d, EEE, ''yy (HH:mm:ss)", Locale.getDefault())
    }
}
